tests/unittests/Foundation/ReferenceFoundation/TestNSTimeZone.mm

```diff
@@ -0,0 +1,68 @@
+//******************************************************************************
+//
+// Copyright (c) 2016 Microsoft Corporation. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2015 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+
+#import <Foundation/Foundation.h>
+#import <TestFramework.h>
+#import <windows.h>
+
+#import <unicode/timezone.h>
+
+TEST(NSTimeZone, Abbreviation) {
+    auto tz = [NSTimeZone systemTimeZone];
+    auto abbreviation1 = tz.abbreviation;
+    auto abbreviation2 = [tz abbreviationForDate:[NSDate date]];
+    ASSERT_OBJCEQ_MSG(abbreviation1, abbreviation2, @"%@ should be equal to %@", abbreviation1, abbreviation2);
+}
+
+TEST(NSTimeZone, InitializingTimeZoneWithOffset) {
+    NSTimeZone* tz = [NSTimeZone timeZoneWithName:@"GMT-0400"];
+    ASSERT_OBJCNE(nil, tz);
+    auto seconds = [tz secondsFromGMTForDate:[NSDate date]];
+    ASSERT_EQ_MSG(seconds, -14400, @"GMT-0400 should be -14400 seconds but got %d instead", seconds);
+}
+
+TEST(NSTimeZone, InitializingTimeZoneWithAbbreviation) {
+    // Test invalid timezone abbreviation
+    NSTimeZone* tz = [NSTimeZone timeZoneWithAbbreviation:@"XXX"];
+    ASSERT_OBJCEQ(nil, tz);
```

Comment:
>ASSERT_OBJCEQ [](start = 4, length = 13)

Since the AST test isn't contingent upon the result of this one, you should use the `EXPECT` version of the macro instead.
(Applies globally.)